src/m365/teams/commands/tab/tab-get.ts

```diff
@@ -0,0 +1,257 @@
+        }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple Microsoft Teams teams with name ${args.options.teamName} found: ${response.value.map(x => x.id)}`);
+        }
+        
+        return Promise.resolve(teamItem.id);
+      });
+  }
+
+  private getChannelId(args: CommandArgs): Promise<string> {
+    if (args.options.channelId) {
+      return Promise.resolve(args.options.channelId);
+    }
+
+    const channelRequestOptions: any = {
+      url: `${this.resource}/v1.0/teams/${encodeURIComponent(this.teamId)}/channels?$filter=displayName eq '${encodeURIComponent(args.options.channelName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      json: true
+    };
+
+    return request.get<{ value: Channel[] }>(channelRequestOptions)
+      .then(response => {
+        const channelItem: Channel | undefined = response.value[0];
+
+        if (!channelItem) {
+          return Promise.reject(`The specified channel does not exist in the Microsoft Teams team`);
+        }
+        
+        return Promise.resolve(channelItem.id);
+      });
+  }
+
+  private getTabId(args: CommandArgs): Promise<string> {
+    if (args.options.tabId) {
+      return Promise.resolve(args.options.tabId);
+    }
+
+    const tabRequestOptions: any = {
+      url: `${this.resource}/v1.0/teams/${encodeURIComponent(this.teamId)}/channels/${encodeURIComponent(this.channelId)}/tabs?$filter=displayName eq '${encodeURIComponent(args.options.tabName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      json: true
+    }
+
+    return request.get<{ value: Tab[] }>(tabRequestOptions)
+    .then(response => {
+      const tabItem: Tab | undefined = response.value[0];
+
+      if (!tabItem) {
+        return Promise.reject(`The specified tab does not exist in the Microsoft Teams team channel`);
+      }
+
+      return Promise.resolve(tabItem.id);
+    });
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    this
+      .getTeamId(args)
+      .then((teamId: string): Promise<string> => {
+        this.teamId = teamId;
+        return this.getChannelId(args);
+      })
+      .then((channelId: string): Promise<string> => {
+        this.channelId = channelId;
+        return this.getTabId(args);
+      })
+      .then((tabId: string): Promise<Tab> => {
+        const endpoint: string = `${this.resource}/v1.0/teams/${encodeURIComponent(this.teamId)}/channels/${encodeURIComponent(this.channelId)}/tabs/${encodeURIComponent(tabId)}`;
+
+        const requestOptions: any = {
+          url: endpoint,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          json: true
+        }
+
+        return request.get<Tab>(requestOptions);
+      })
+      .then((res: Tab): void => {        
+        cmd.log(res.webUrl);
```

Comment:
Since the command is named `teams tab get` rather than `teams tab url get`, let's output all properties to match the command's purpose as defined in the description.